Automattic/wp-calypso
# Gutenberg: Add Document Title to the Gutenberg Editor

Fix #27300 

Add a document title to the Gutenberg editor based on the post type, and if the post is new or not.

This new component is heavily inspired by the [classic editor version](https://github.com/Automattic/wp-calypso/blob/master/client/post-editor/editor-document-head/index.jsx), although it's slightly simplified, as the Gutenberg editor already has all all the needed info available.

For example:
- `QueryPostTypes` is already called by the [editor itself](https://github.com/Automattic/wp-calypso/blob/master/client/gutenberg/editor/main.jsx#L43), and there is no need to call it again.
- The `postId` presence is enough, for now, to determine if the editor is a New or Edit one, but this will likely change after https://github.com/Automattic/wp-calypso/pull/27301 is merged. I'll rebase and update when it happens.

## Testing instructions

Visit http://calypso.localhost:3000/gutenberg for the following routes and make sure the document title is the expected one.

| Route | Title |
| - | - |
| `/post/{ site }` | New Post |
| `/post/{ site }/{ postId }` | Edit Post |
| `/page/{ site }` | New Page |
| `/page/{ site }/{ postId }` | Edit Page |
| `/edit/{ customPostType }/{ site }` | { customPostType.labels.new_item } (e.g. New Post) |
| `/edit/{ customPostType }/{ site }/{ postId }` | { customPostType.labels.edit_item } (e.g. Edit Post) |

Note that when loading a CPT, the document title should "flash" from `New` (or `Edit`) to `New CPT` (or `Edit CPT`), and that most CPT are indeed supposed to be labelled as `Post` (e.g. `New Post` or `Edit Post`).
